include/circt/Dialect/FIRRTL/FIRRTLOps.h

```diff
@@ -34,6 +34,12 @@ bool isExpression(Operation *op);
 /// Return the number of ports in a module-like thing (modules, memories, etc)
 size_t getNumPorts(Operation *op);
 
+/// Return true if the specified operation has a constant value. This trivially
+/// checks for `firrtl.constant` and friends, but also looks through subaccesses
+/// and correctly handles wires driven with only constant values.
+bool isConstant(Operation *op);
+bool isConstant(Value value);
```

Comment:
Does the comment get applied to both methods in doxygen?

Reply:
I don't think so. However, it retains the order in the output AFAIK, and I've seen LLVM code use this pattern as well. Not really nice, though :-/

Reply:
Sounds good to me.